luacompiler/src/lib/frontend/mod.rs

```diff
@@ -0,0 +1,174 @@
+                    let new_var = reg_map.new_reg();
+                    instrs.push(self.get_instr(&nodes[1], new_var, left, right));
+                    new_var
+                }
+            }
+            Term { lexeme } => {
+                let value = self.pt.get_string(lexeme.start(), lexeme.end());
+                match lexeme.tok_id() {
+                    lua5_3_l::T_NUMERAL => {
+                        let reg = reg_map.new_reg();
+                        if value.contains(".") {
+                            let fl = const_map.get_float(value.to_string());
+                            instrs.push(make_instr(Opcode::LDF, reg, fl, 0));
+                        } else {
+                            let int = const_map.get_int(value.parse().unwrap());
+                            instrs.push(make_instr(Opcode::LDI, reg, int, 0));
+                        }
+                        reg
+                    }
+                    lua5_3_l::T_SHORT_STR => {
+                        let reg = reg_map.new_reg();
+                        let len = value.len();
+                        // make sure that the quotes are not included!
+                        let short_str = const_map.get_str(value[1..(len - 1)].to_string());
+                        instrs.push(make_instr(Opcode::LDS, reg, short_str, 0));
+                        reg
+                    }
+                    _ => reg_map.get_reg(value),
+                }
+            }
+        }
+    }
+
+    /// Get the appropriate instruction for a given Node::Term.
+    fn get_instr(&self, node: &Node<u8>, reg: u8, lreg: u8, rreg: u8) -> u32 {
+        if let Term { lexeme } = node {
+            let opcode = match lexeme.tok_id() {
+                lua5_3_l::T_PLUS => Opcode::ADD,
+                lua5_3_l::T_MINUS => Opcode::SUB,
+                lua5_3_l::T_STAR => Opcode::MUL,
+                lua5_3_l::T_FSLASH => Opcode::DIV,
+                lua5_3_l::T_MOD => Opcode::MOD,
+                lua5_3_l::T_FSFS => Opcode::FDIV,
+                lua5_3_l::T_CARET => Opcode::EXP,
+                _ => unimplemented!("Instruction {}", lexeme.tok_id()),
+            };
+            make_instr(opcode, reg, lreg, rreg)
+        } else {
+            panic!("Expected a Node::Term!");
+        }
+    }
+
+    /// Find the first Node::Term with the given id.
+    fn find_term(start: &Node<u8>, id: u8) -> Option<&Node<u8>> {
+        let mut pt_nodes: Vec<&Node<u8>> = vec![start];
+        while !pt_nodes.is_empty() {
+            let node = pt_nodes.pop().unwrap(); // always checked if it is empty
+            match node {
+                Nonterm { ridx: _, ref nodes } => {
+                    for ref node in nodes {
+                        pt_nodes.push(node);
+                    }
+                }
+                Term { lexeme } => {
+                    if lexeme.tok_id() == id {
+                        return Some(node);
+                    } else {
```

Comment:
The `else` branch here seems entirely pointless?

Reply:
that's a good point